provider/server/server.go

```diff
@@ -68,9 +72,24 @@ func setOptionsDefaults(o *Options) {
 	}
 }
 
+func convertGuestAgentFromString(guestAgent string) (api.GuestAgent, error) {
+	availableGuestAgents := api.GuestAgentAvailable()
+	index := slices.Index(availableGuestAgents, guestAgent)
+	if index == -1 {
+		return api.GuestAgentNone, fmt.Errorf("unsupported guest agent type: %s", guestAgent)
+	}
+
+	return api.GuestAgent(availableGuestAgents[index]), nil
+}
+
 func New(opts Options) (*Server, error) {
 	setOptionsDefaults(&opts)
 
+	guestAgent, err := convertGuestAgentFromString(opts.GuestAgent)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
wdyt about passing the `GuestAgent` here and to the validation outside of the server?

Reply:
GuestAgent is currently used in server only. If you want to implement conversion somewhere else, it can be do, but we can follow my suggestion with implement pflag.Value interface because in other case we will end with multiple convert functions this way and implementation of pflag.Value is fast failure.

Reply:
if you don't like use API type for that. We can move GuestAgent into new package. Maybe something as features/guestagent ? And it can be import into provider/cmd/app and pkg/api as independent package and we can without big problems implement pflag.Value interface

Reply:
reimplemented as pflag.Value interface:

```
Error: invalid argument "Test" for "--guest-agent-type" flag: unsupported option Test
Usage:
  libvirt-provider [flags]

Flags:
      --address string                             Address to listen on. (default "/var/run/iri-machinebroker.sock")
      --apinet-kubeconfig string                   Path to the kubeconfig file for the apinet-cluster.
      --apinet-node-name string                    APInet node name
```

Reply:
fixed